kernel_gateway/services/kernels/handlers.py

```diff
@@ -19,6 +20,7 @@ class MainKernelHandler(TokenAuthorizationMixin,
     """Extends the notebook main kernel handler with token auth, CORS, and
     JSON errors.
     """
+    @gen.coroutine
     def post(self):
```

Comment:
Thanks! The reason it works now is probably because we don't do any async operations in the default implementation of these calls. They are made async to allow for slower implementations to be used.